### PR TITLE
logseq: 0.10.1 -> 0.10.3

### PR DESCRIPTION
**Pull request by kilianar**
## Description of changes
- https://github.com/logseq/logseq/releases/tag/0.10.2
- https://github.com/logseq/logseq/releases/tag/0.10.3

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by dixslyf** [comment]
1 package built:
logseq

Result of `nixpkgs-review pr 274180` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>logseq</li>
  </ul>
</details>

**Review by dixslyf — APPROVED** [review]
Ran it for a bit and I don't see any issues. LGTM.

**Comment by schmittlauch** [comment]
Unfortunately, this re-opens #245586 for me, this time with NODE_MODULE_VERSION 116 vs. 118:

```
23:34:12.027 › Error: The module '/nix/store/j87nnq8q2b294b87gn3qma2yhn3qkv8r-logseq-0.10.1/share/logseq/resources/app/node_modules/better-sqlite3/build/Release/better_sqlite3.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 116. This version of Node.js requires
NODE_MODULE_VERSION 118. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).: logseq_local_/home/spiollinux/Seafile/clientsync/notes/Logseq
23:34:12.028 › cannot unlink search db: Error: ENOENT: no such file or directory, unlink '/home/spiollinux/.config/Logseq/search/logseq_local__home_spiollinux_Seafile_clientsync_notes_Logseq'
    at Object.unlinkSync (node:fs:1883:3)
    at $electron$search$open_db_BANG_$$ (/nix/store/j87nnq8q2b294b87gn3qma2yhn3qkv8r-logseq-0.10.1/share/logseq/resources/app/electron.js:14510:42)
    at $electron$search$get_or_open_db$$ (/nix/store/j87nnq8q2b294b87gn3qma2yhn3qkv8r-logseq-0.10.1/share/logseq/resources/app/electron.js:14552:3)
    at $electron$search$upsert_blocks_BANG_$$ (/nix/store/j87nnq8q2b294b87gn3qma2yhn3qkv8r-logseq-0.10.1/share/logseq/resources/app/electron.js:14582:41)
    at /nix/store/j87nnq8q2b294b87gn3qma2yhn3qkv8r-logseq-0.10.1/share/logseq/resources/app/electron.js:15488:63
    at $JSCompiler_prototypeAlias$$.$cljs$core$IFn$_invoke$arity$2$ (/nix/store/j87nnq8q2b294b87gn3qma2yhn3qkv8r-logseq-0.10.1/share/logseq/resources/app/electron.js:9476:178)
    at /nix/store/j87nnq8q2b294b87gn3qma2yhn3qkv8r-logseq-0.10.1/share/logseq/resources/app/electron.js:16076:76
    at WebContents.<anonymous> (node:electron/js2c/browser_init:2:88656)
    at WebContents.emit (node:events:514:28)
23:34:12.634 › Error: The module '/nix/store/j87nnq8q2b294b87gn3qma2yhn3qkv8r-logseq-0.10.1/share/logseq/resources/app/node_modules/better-sqlite3/build/Release/better_sqlite3.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 116. This version of Node.js requires
NODE_MODULE_VERSION 118. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).: logseq_local_/home/spiollinux/Seafile/clientsync/notes/Logseq
23:34:12.635 › cannot unlink search db: Error: ENOENT: no such file or directory, unlink '/home/spiollinux/.config/Logseq/search/logseq_local__home_spiollinux_Seafile_clientsync_notes_Logseq'
    at Object.unlinkSync (node:fs:1883:3)
    at $electron$search$open_db_BANG_$$ (/nix/store/j87nnq8q2b294b87gn3qma2yhn3qkv8r-logseq-0.10.1/share/logseq/resources/app/electron.js:14510:42)
    at $electron$search$get_or_open_db$$ (/nix/store/j87nnq8q2b294b87gn3qma2yhn3qkv8r-logseq-0.10.1/share/logseq/resources/app/electron.js:14552:3)
    at $electron$search$upsert_pages_BANG_$$ (/nix/store/j87nnq8q2b294b87gn3qma2yhn3qkv8r-logseq-0.10.1/share/logseq/resources/app/electron.js:14556:41)
    at /nix/store/j87nnq8q2b294b87gn3qma2yhn3qkv8r-logseq-0.10.1/share/logseq/resources/app/electron.js:15501:62
    at $JSCompiler_prototypeAlias$$.$cljs$core$IFn$_invoke$arity$2$ (/nix/store/j87nnq8q2b294b87gn3qma2yhn3qkv8r-logseq-0.10.1/share/logseq/resources/app/electron.js:9476:178)
    at /nix/store/j87nnq8q2b294b87gn3qma2yhn3qkv8r-logseq-0.10.1/share/logseq/resources/app/electron.js:16076:76
    at WebContents.<anonymous> (node:electron/js2c/browser_init:2:88656)
    at WebContents.emit (node:events:514:28)
Warning: vkCreateInstance: Found no drivers!
Warning: vkCreateInstance failed with VK_ERROR_INCOMPATIBLE_DRIVER
    at CheckVkSuccessImpl (../../third_party/dawn/src/dawn/native/vulkan/VulkanError.cpp:88)
    at CreateVkInstance (../../third_party/dawn/src/dawn/native/vulkan/BackendVk.cpp:458)
    at Initialize (../../third_party/dawn/src/dawn/native/vulkan/BackendVk.cpp:344)
    at Create (../../third_party/dawn/src/dawn/native/vulkan/BackendVk.cpp:266)
    at operator() (../../third_party/dawn/src/dawn/native/vulkan/BackendVk.cpp:521)
```

**Comment by dixslyf** [comment]
> Unfortunately, this re-opens #245586 for me, this time with NODE_MODULE_VERSION 116 vs. 118:

I can confirm the same issue on my machine.

Apologies for the oversight.

(If someone with write access could dismiss my previous review, that would be great.)

**Comment by schmittlauch** [comment]
@dixslyf I think you could dismiss your own review – as a workaround I just requested your re-review though.

Regarding the way forward, my rough understanding of the situation: As we use the binary upstream pre-compiled AppImage as of now, we are bound to the compiled better-sqlite3 version shipped in that image. Thus we could:

- wait for upstream to also build their AppImage against electron 28 with node module 118 API
- try to monkey-/ binary patch that image somehow
- fully build logseq from source.

It'd be awesome for a source-based distro to achieve the latest, but I myself have exactly zero experience of how to build Electron stuff in Nix the least painful way.

**Comment by schmittlauch** [comment]
@delroth Is you adding the "approvals: 1" label an automated process? I guess so, but here I deliberately removed that label and did my best to dismiss the review, as the reviewer themself wanted to retract their approval.

**Comment by kilianar** [comment]
The issue with the mismatched node module versions didn't arise during my brief testing of the update. It would be advantageous to build logseq from source, yet this appears to be a more extensive task. Therefore, I will convert this pull request into a draft for the time being and revise it once a new release becomes available.

**Review by CactiChameleon9 — COMMENTED** [review]
Seems completely broken for me?

```
daniel@danielLaptop ~> logseq
Error launching app
Unable to find Electron app at /nix/store/zb3cs3g1027kdwgchn0qf7vdar7l74pw-logseq-0.10.2/share/logseq/resources/app

Cannot find module '/nix/store/zb3cs3g1027kdwgchn0qf7vdar7l74pw-logseq-0.10.2/share/logseq/resources/app'
Require stack:
- /nix/store/bs4z1qwgkvn8y4n2b0g4imkxfkqcgnpl-electron-unwrapped-27.1.3/libexec/electron/resources/default_app.asar/main.js
- 
```

I have `/nix/store/zb3cs3g1027kdwgchn0qf7vdar7l74pw-logseq-0.10.2/share/logseq/resources/app.asar.unpacked/node_modules/` existing but not the mentioned app folder

File: pkgs/applications/misc/logseq/default.nix
Comment (jtrees) [inline pkgs/applications/misc/logseq/default.nix:6]: Should we move this comment to `all-package.nix` now? I think it'll be most useful where the version is being pinned.

**Review by jtrees — APPROVED** [review]
Code looks fine to me and everything seems to work. Didn't run into the NODE_MODULE_VERSION errors others mentioned.

Thanks!

**Comment by jtrees** [comment]
Result of `nixpkgs-review pr 274180` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>logseq</li>
  </ul>
</details>

**Review by schmittlauch — APPROVED** [review]
I've triggered a reindexing of my graph, the previous sqlite errors do __not__ appear again.

LGTM.

**Review by CactiChameleon9 — APPROVED** [review]

**Comment by greg-hellings** [comment]
Because no good deed goes unpunished, 0.10.3 was released earlier today.

**Comment by dixslyf** [comment]
> @dixslyf I think you could dismiss your own review – as a workaround I just requested your re-review though.

I thought so as well, but apparently GitHub only allows repo admins and people with write access to dismiss reviews.

**Review by dixslyf — APPROVED** [review]
PR looks good to me. No more errors.

1 package built:
logseq

Result of `nixpkgs-review pr 274180` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>logseq</li>
  </ul>
</details>

File: pkgs/applications/misc/logseq/default.nix
Comment (kilianar) [reply pkgs/applications/misc/logseq/default.nix]: I overlooked that one, but it's fixed now. Thank you!

**Review by schmittlauch — APPROVED** [review]

**Review by dixslyf — APPROVED** [review]
Electron 27.2.0 has finally made it to the binary cache.

Triggered a re-index of my graph and I don't see any errors. LGTM.

---

Result of `nixpkgs-review pr 274180` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>logseq</li>
  </ul>
</details>

File: pkgs/applications/misc/logseq/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/misc/logseq/default.nix:6]: Is the comment still correct? If so we should probably keep it and continue to track the major version.

File: pkgs/top-level/all-packages.nix
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:3184]: We should probably move that back. It is also more cross friendly to keep this in the file.

File: pkgs/applications/misc/logseq/default.nix
Comment (kilianar) [reply pkgs/applications/misc/logseq/default.nix]: Since I don't utilize the syncing feature, I am unsure.

**Review by wegank — APPROVED** [review]

File: pkgs/applications/misc/logseq/default.nix
Comment (dixslyf) [reply pkgs/applications/misc/logseq/default.nix]: @hall Since you wrote the original comment, perhaps you can verify this for us?

Comment (jocelynthode) [reply pkgs/applications/misc/logseq/default.nix]: @SuperSandro2000 It looks like this was changed to `electron_27`. Is this now mergeable?

Comment (wegank) [reply pkgs/applications/misc/logseq/default.nix]: Checked that upstream moved to Electron 27: https://github.com/logseq/logseq/commit/de13df47229656d9996c9e28c5323f1b064f86d1

**Pull request merged**
